### PR TITLE
Allow to pass a callable fucnction as window argument

### PR DESCRIPTION
**Pull request by FranzForstmayr**
This is useful to define more customized windows and not just the window functions from scipy.

### CODE REVIEW

**Comment by FranzForstmayr** [comment]
could we get this into #924 ? :)

File: skrf/time.py [hunk 1]
```diff
@@ -225,7 +225,22 @@ def detect_span(ntwk: 'Network', t_unit: str = "") -> float:
...
+    """Calls a custom window function or `scipy.signal.get_window()` depending on the window argument.
 
+    Args:
+        window (Union[str, tuple, Callable]): The callable window function or a valid window string
+            for `scipy.signal.get_window()`.
+        Nx (int): Number of samples
+
+    Returns:
+        npy.ndarray: Window samples
+    """
```
Comment (jhillairet) [inline skrf/time.py:238]: switch to Numpy style docstring:
```suggestion
    Parameters
    ----------
    window : str, tuple or Callable
        The callable window function or a valid window string
        for `scipy.signal.get_window()`.
    Nx : int
        Number of samples

    Returns
    -------
    window : npy.ndarray
        Window samples.
    """
```

**Review by jhillairet — APPROVED** [review]
sure we can include it in the next release.

**Review by jhillairet — APPROVED** [review]

**Comment by FranzForstmayr** [comment]
The function is currently not in the public API, do we need this for this case?
https://scikit-rf--925.org.readthedocs.build/en/925/api/time.html

**Comment by jhillairet** [comment]
Eventually. You can add the function name in the docstring at the top of time.py

**Review by FranzForstmayr — COMMENTED** [review]
Add `get_window` to sphinx doc

**Pull request merged**